src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -1747,33 +1747,47 @@ public LazyFixedStatement(Lazy<IVariableDeclarationStatement> variables, Lazy<IO
     }
 
     /// <summary>
-    /// Represents a C# foreach statement or a VB For Each statement.
+    /// Represents a C# 'foreach' statement or a VB 'For Each' statement.
     /// </summary>
     internal abstract partial class BaseForEachLoopStatement : LoopStatement, IForEachLoopStatement
     {
-        public BaseForEachLoopStatement(ILocalSymbol iterationVariable, LoopKind loopKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
-            base(loopKind, OperationKind.LoopStatement, semanticModel, syntax, type, constantValue, isImplicit)
+        public BaseForEachLoopStatement(ImmutableArray<ILocalSymbol> locals, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
+            base(LoopKind.ForEach, locals, OperationKind.LoopStatement, semanticModel, syntax, type, constantValue, isImplicit)
         {
-            IterationVariable = iterationVariable;
         }
-        /// <summary>
-        /// Iteration variable of the loop.
-        /// </summary>
-        public ILocalSymbol IterationVariable { get; }
+        protected abstract IOperation LoopControlVariableImpl { get; }
         protected abstract IOperation CollectionImpl { get; }
+        protected abstract ImmutableArray<IOperation> NextVariablesImpl { get; }
         public override IEnumerable<IOperation> Children
         {
             get
             {
+                if (LoopControlVariable != null)
+                {
+                    yield return LoopControlVariable;
```

Comment:
>yield return LoopControlVariable; [](start = 20, length = 33)

Are children supposed to be returned in any particular order?

Reply:
We do not have any specific guarantee or principles for order of children nodes right now. But we should discuss this at the design meeting. Filed https://github.com/dotnet/roslyn/issues/22020